library/Fission/Internal/Development.hs

```diff
@@ -0,0 +1,127 @@
+
+     == Example Use
+
+     > runOne Fission.IPFS.Peer.all
+     > -- Right ["/ip4/3.215.160.238/tcp/4001/ipfs/QmVLEz2SxoNiFnuyLpbXsH6SvjPTrHNMU88vCQZyhgBzgw"]
+-}
+runOne :: RIO Config a -> IO a
+runOne action = do
+  logOptions   <- logOptionsHandle stdout True
+  withLogFunc (setLogUseTime True logOptions) $ \_logFunc -> do
+    _dbPool      <- runSimpleApp $ connPool 1 1 3600 pgConnectInfo
+    _processCtx  <- mkDefaultProcessContext
+    _httpManager <- HTTP.newManager HTTP.defaultManagerSettings
+    run _logFunc _dbPool _processCtx _httpManager action
+
+{- | Run some action(s) in the app's context,
+     but asks for existing portions of the setup that require side effects,
+     in case they're already available (which is more efficient).
+
+     == Example Use
+
+     > dbPool <- runSimpleApp $ connPool 1 1 3600 _pgConnectInfo
+     > processCtx <- mkDefaultProcessContext
+     > httpManager <- HTTP.newManager HTTP.defaultManagerSettings
+     > logOptions <- logOptionsHandle stdout True
+     > (logFunc, _ :: IO ()) <- newLogFunc $ setLogUseTime True logOptions
+     >
+     > let runSession = run logFunc dbPool processCtx httpManager
+     >
+     > runSession Fission.IPFS.Peer.all
+     > -- Right ["/ip4/3.215.160.238/tcp/4001/ipfs/QmVLEz2SxoNiFnuyLpbXsH6SvjPTrHNMU88vCQZyhgBzgw"]
+     >
+     > runSession Fission.IPFS.Peer.connect Fission.peer
+     > -- ()
+-}
+run :: LogFunc
+    -> DB.Pool
+    -> ProcessContext
+    -> HTTP.Manager
+    -> RIO Config a
+    -> IO a
+run _logFunc _dbPool _processCtx _httpManager action =
+  runRIO config do
+    logDebug $ displayShow config
+    action
+  where
+    config = Config {..}
+
+    _herokuID       = Hku.ID       "HEROKU_ID"
+    _herokuPassword = Hku.Password "HEROKU_PASSWORD"
+
+    _ipfsPath    = "/usr/local/bin/ipfs"
+    _ipfsURL     = IPFS.URL $ BaseUrl Http "localhost" 5001 ""
+    _ipfsTimeout = IPFS.Timeout 3600
+
+    _pgConnectInfo = pgConnectInfo
+
+    _host = "mycoolapp.io"
+
+{- | Setup a complete development configuration.
+
+     Note that this does not clean up the log function,
+     but does return an action to do so.
+
+     == Example Use
+
+     > (cfg, _) <- mkConfig
+     > let run' = runRIO cfg
+     > run' Fission.IPFS.Peer.all
+     > -- Right ["/ip4/3.215.160.238/tcp/4001/ipfs/QmVLEz2SxoNiFnuyLpbXsH6SvjPTrHNMU88vCQZyhgBzgw"]
+     >
+     > run' Fission.IPFS.Peer.connect Fission.peer
+     > -- ()
+
+     If you need to overwrite any fields: use record update syntax, or the 'Config' lenses.
+
+     > (cfg, _) <- mkConfig
+     > let run' = runRIO cfg { _ipfsPath = "~/Downloads/ipfs" }
+     > run' Fission.IPFS.Peer.all
+     > -- Right ["/ip4/3.215.160.238/tcp/4001/ipfs/QmVLEz2SxoNiFnuyLpbXsH6SvjPTrHNMU88vCQZyhgBzgw"]
+-}
+mkConfig :: IO (Config, IO ())
+mkConfig = do
+  let
+    _herokuID       = Hku.ID       "HEROKU_ID"
+    _herokuPassword = Hku.Password "HEROKU_PASSWORD"
+    _ipfsPath       = "/usr/local/bin/ipfs"
```

Comment:
Seeing a bunch of reused values for config in both `run` and `mkConfig` could we not move them to one record/function called by each?

Reply:
Yep, sure can! Given that they're placeholders for development, I didn't go back and clean them up, but yeah, fewer surprises are good. The easiest thing would probably be to break them into their own definitions in this module.

```haskell
ipfsPath :: FilePath
ipfsPath = "/usr/local/bin/ipfs" 
```

That kind of thing

Reply:
Okay, broke this out into `mkConfig` and `mkConfig'`

Convention: `foo'` is either used for inner values, or as "same as the non-tick version, but with some defaults set for you if you're feeling lazy".

Also `foo_` is convention for "foo, but we throw the result away".

Reply:
Hey @expede thanks for breaking those out I had more meant overall across each function in this file

For example `mkConfig` and `run` both use the following values
```haskell
    _herokuID       = Hku.ID       "HEROKU_ID"
    _herokuPassword = Hku.Password "HEROKU_PASSWORD"
    _ipfsPath    = "/usr/local/bin/ipfs"
    _ipfsURL     = IPFS.URL $ BaseUrl Http "localhost" 5001 ""
    _ipfsTimeout = IPFS.Timeout 3600
    _pgConnectInfo = pgConnectInfo
    _host = "mycoolapp.io"
```

could we not extract each of these the same way we do for `pgConnectInfo`

```haskell
pgConnectInfo :: PGConnectInfo
pgConnectInfo = PGConnectInfo "localhost" 5432 "web_api" Nothing Nothing Nothing

ipfsPath :: FilePath
ipfsPath = "/usr/local/bin/ipfs" 

-- &c
```